### PR TITLE
feat: allow route to inherit hosts from service

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by tokers — APPROVED** [review]

File: t/router/radixtree-host-uri2.t [hunk 1]
```diff
@@ -376,3 +376,39 @@ GET /server_port
...
+
+
+
+=== TEST 14: inherit hosts from services
+--- yaml_config eval: $::yaml_config
+--- apisix_yaml
+services:
+  - id: 1
+    hosts:
+      - bar.com
```
Comment (tzssangglass) [inline t/router/radixtree-host-uri2.t:388]: the service hosts here is `bar.com`, but the host in request is `www.foo.com`,how they match?

File: t/router/radixtree-host-uri3.t [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+            if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+            ngx.sleep(0.1)
+
+            local httpc = http.new()
+            local res, err = httpc:request_uri(uri, {headers = {Host = "foo.com"}})
```
Comment (nic-chen) [inline t/router/radixtree-host-uri3.t:114]: should be better if we have another test case without `Host`

File: t/router/radixtree-host-uri2.t (hunk 1 above)
Comment (spacewander) [reply t/router/radixtree-host-uri2.t:388]: Yes, we test with unmatched. If the host doesn't take effect, the route will match and the response body is "hello1 world".

File: t/router/radixtree-host-uri3.t (hunk 2 above)
Comment (spacewander) [reply t/router/radixtree-host-uri3.t]: Added.

**Review by nic-chen — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Comment by Donghui0** [comment]
in `plugin.merge_service_route` function,  need to add processing for route.value.timeout. Otherwise, the timeout attribute of ApisixRoute will be invalid.

I have a question in the `merge service_route` method. Why not directly core.table.deepcopy(route_conf), assign the attribute of service to new_conf ?

**Pull request merged**